edx_sga/sga.py

```diff
@@ -221,17 +234,24 @@ def save_sga(self, data, suffix=''):
 
     @XBlock.handler
     def upload_assignment(self, request, suffix=''):
-        # pylint: disable=unused-argument
+        # pylint: disable=unused-argument, protected-access
         """
         Save a students submission file.
         """
         require(self.upload_allowed())
         user = self.get_real_user()
         require(user)
+        upload = request.params['assignment']
+        if self.file_size_over_limit(upload.file):
+            raise JsonHandlerError(
+                413, 'Unable to upload file. Max size limit is {size}'.format(
+                    size=self.STUDENT_FILEUPLOAD_MAX_SIZE
```

Comment:
`file_size_over_limit` prefers the Django settings value for `STUDENT_FILEUPLOAD_MAX_SIZE` over the `STUDENT_FILEUPLOAD_MAX_SIZE` class property. If that Django settings value is set, then you're returning the wrong upload limit here. I think it makes sense to rename the `STUDENT_FILEUPLOAD_MAX_SIZE` property to `DEFAULT_STUDENT_FILEUPLOAD_MAX_SIZE`, create a new `@property` called `student_upload_max_size` or something like that, and implement it like this:

```python
@property
def student_upload_max_size(self):
    return getattr(
        settings, 
        "STUDENT_FILEUPLOAD_MAX_SIZE",
        self.DEFAULT_STUDENT_FILEUPLOAD_MAX_SIZE
    )
```

Then `file_size_over_limit` and this handler method would both read `self.student_upload_max_size` to determine the effective max upload size.